### DISCUSSION TITLE
c vs binary package

### DISCUSSION THREAD

**Discussion by AndrewJackson2020** [psycopg/psycopg#971] [Q&A]
Trying to figure out the upsides and downsides of the c and binary packaging. According to the documentation:

```
The library will include the speed-up C module and will be linked to the system libraries (libpq, libssl…) so that system upgrade of libraries will upgrade the libraries used by Psycopg 3 too. This is the preferred way to install Psycopg for a production site.
```

Is the only disadvantage of using the binary installation that the install won't keep up to date with the system libpq, libssl, etc? Is there a risk to this? Are there other disadvantages to using the binary installation (stability, performance, etc.). 

I also found this discussion where it is mentioned that there might be errors in multi threaded environments though this may no longer apply to psycopg.

https://github.com/psycopg/psycopg2/issues/1283

**Comment by dvarrazzo** [comment]
binary and c are exactly the same code:

- `binary` is compiled by us, and packaged with a set of libraries our side
- `c` is compiled by you, and dynamically linked with your libraries.

The problems of stability of the psycopg2-binary package were, for the 95% of the part, related to the very old versions of libpq we were packaging at time. There are other sporadic reports of the wheel not working for specific cases; I have no immediate case in mind, maybe [some of the past tickets](https://github.com/psycopg/psycopg2/issues?q=label%3Awheel) might reveal problems you can relate.

So I don't see immediate risks. The `binary` package is a convenience that we provide on a best-effort basis, which may be suitable to you or not, but I don't see specific risks in using it.
